apps/web-giddh/src/app/shared/eWayBill/eWayBill/eWayBill.component.ts

```diff
@@ -0,0 +1,769 @@
+        transMode: null,
+        vehicleType: null,
+    };
+    @ViewChild(BsDatepickerDirective, { static: true }) public datepickers: BsDatepickerDirective;
+    public selectedEway: Result;
+    public states: any[] = [];
+    /** directive to get reference of element */
+    @ViewChild('datepickerTemplate') public datepickerTemplate: TemplateRef<any>;
+    /* This will store selected date range to use in api */
+    public selectedDateRange: any;
+    /* This will store selected date range to show on UI */
+    public selectedDateRangeUi: any;
+    /* This will store available date ranges */
+    public datePickerOption: any = GIDDH_DATE_RANGE_PICKER_RANGES;
+    /* dayjs object */
+    public dayjs = dayjs;
+    /* Selected from date */
+    public fromDate: string;
+    /* Selected to date */
+    public toDate: string;
+    /* Selected range label */
+    public selectedRangeLabel: any = "";
+    /* Universal date observer */
+    public universalDate$: Observable<any>;
+    /* This will store the x/y position of the field to show datepicker under it */
+    public dateFieldPosition: any = { x: 0, y: 0 };
+    private destroyed$: ReplaySubject<boolean> = new ReplaySubject(1);
+    /* This will hold local JSON data */
+    public localeData: any = {};
+    /* This will hold common JSON data */
+    public commonLocaleData: any = {};
+    /** True if today selected */
+    public todaySelected: boolean = false;
+    /** True if dropdown menu needs to show upwards */
+    public isDropUp: boolean = false;
+    /** Observable to store the branches of current company */
+    public currentCompanyBranches$: Observable<any>;
+    /** Stores the branch list of a company */
+    public currentCompanyBranches: Array<any>;
+    /** Stores the current branch */
+    public currentBranch: any = { name: '', uniqueName: '' };
+    /** Stores the current organization type */
+    public currentOrganizationType: OrganizationType;
+    /** Stores the current company */
+    public activeCompany: any;
+    /** This will be hold for universal date */
+    public universalDate: any[] = [];
+    /** True if initial api got called */
+    public initialApiCalled: boolean = false;
+    /** Stores the tax details of a company */
+    public taxes: IOption[] = [];
+    /** Holds active tab index */
+    public activeTabIndex: number = 0;
+    /** Stores the selected tab */
+    public selectedTab: string;
+    /** Stores the displayed columns */
+    public displayedColumns: string[] = ['index', 'invoiceDate', 'docNumber', 'customerName', 'customerGstin', 'ewbNo', 'ewayBillDate', 'totalValue', 'actions'];
+    /** Stores the cancel dialog reference */
+    public cancelDialogRef: MatDialogRef<any>;
+    /** True if consolidated branch */
+    public isConsolidatedBranch: boolean;
+
+    constructor(
+        private store: Store<AppState>,
+        private invoiceActions: InvoiceActions,
+        private _invoiceService: InvoiceService,
```

Comment:
```suggestion
        private invoiceService: InvoiceService,
```

Make this change for all and also check single char variables